clients/python/tensorzero/deprecated.py

```diff
@@ -0,0 +1,134 @@
+"""
+Deprecated type aliases for backward compatibility.
+
+This module contains deprecated types that have been renamed.
+Import from the main tensorzero module instead of using the old names directly.
+"""
+
+import warnings
+from typing import Any
+
+from typing_extensions import deprecated
+
+from .generated_types import (
+    InferenceFilter,
+    InferenceFilterAnd,
+    InferenceFilterBooleanMetric,
+    InferenceFilterFloatMetric,
+    InferenceFilterNot,
+    InferenceFilterOr,
+    InferenceFilterTag,
+    InferenceFilterTime,
+    InputContentBlock,
+    InputContentBlockFile,
+    InputContentBlockRawText,
+    InputContentBlockTemplate,
+    InputContentBlockText,
+    InputContentBlockThought,
+    InputContentBlockToolCall,
+    InputContentBlockToolResult,
+    InputContentBlockUnknown,
+    StoredInputContentBlock,
+    StoredInputContentBlockFile,
+    StoredInputContentBlockRawText,
+    StoredInputContentBlockTemplate,
+    StoredInputContentBlockText,
+    StoredInputContentBlockThought,
+    StoredInputContentBlockToolCall,
+    StoredInputContentBlockToolResult,
+    StoredInputContentBlockUnknown,
+)
+from .generated_types import (
+    ProviderExtraBody as _ProviderExtraBody,
+)
+from .generated_types import (
+    ProviderExtraBodyDelete as _ProviderExtraBodyDelete,
+)
+from .generated_types import (
+    ProviderExtraHeader as _ProviderExtraHeader,
+)
+from .generated_types import (
+    ProviderExtraHeaderDelete as _ProviderExtraHeaderDelete,
+)
+from .tensorzero import (
+    DICLConfig,
+    DICLOptimizationConfig,
+    RenderedSample,
+)
+from .types import (
+    WorkflowEvaluationRunEpisodeResponse,
+    WorkflowEvaluationRunResponse,
+)
+
+# 2026.1+
+
+DiclConfig = deprecated("Use DICLConfig instead")(DICLConfig)
+
+
+class DiclOptimizationConfig:
+    def __new__(cls, *args: Any, **kwargs: Any):
+        warnings.warn(
+            "Please use `DICLOptimizationConfig` instead of `DiclOptimizationConfig`. In a future release, `DiclOptimizationConfig` will be removed.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return DICLOptimizationConfig(*args, **kwargs)
+
+
+RenderedStoredInference = deprecated("Use RenderedSample instead")(RenderedSample)
+
+InferenceFilterTreeNode = InferenceFilter
+
+FloatMetricFilter = deprecated("Use InferenceFilterFloatMetric instead")(InferenceFilterFloatMetric)
```

Comment:
All of these should be changed too because at runtime the correct type will also have the deprecation warning:

<img width="949" height="229" alt="Image" src="https://github.com/user-attachments/assets/3538ed68-5760-4f72-9319-9d95506dab61" />